packages/utils/interactive-hooks/src/Pressability/Pressability.types.ts

```diff
@@ -74,43 +46,93 @@ export type PressabilityConfig = Readonly<{
    * Called when the hover is deactivated to undo visual feedback.
    */
   onHoverOut?: (event: MouseEvent) => any;
-
-  /**
-   * Called when a long press gesture has been triggered.
-   */
-  onLongPress?: (event: PressEvent) => any;
-
-  /**
-   * Called when a press gestute has been triggered.
-   */
-  onPress?: (event: PressEvent) => any;
-
-  /**
-   * Called when the press is activated to provide visual feedback.
-   */
-  onPressIn?: (event: PressEvent) => any;
-
-  /**
-   * Called when the press location moves. (This should rarely be used.)
-   */
-  onPressMove?: (event: PressEvent) => any;
-
-  /**
-   * Called when the press is deactivated to undo visual feedback.
-   */
-  onPressOut?: (event: PressEvent) => any;
-}>;
-
-export type PressabilityEventHandlers = Readonly<{
-  onBlur: (event: BlurEvent) => void;
-  onClick: (event: PressEvent) => void;
-  onFocus: (event: FocusEvent) => void;
-  onMouseEnter?: (event: MouseEvent) => void;
-  onMouseLeave?: (event: MouseEvent) => void;
-  onResponderGrant: (event: PressEvent) => void;
-  onResponderMove: (event: PressEvent) => void;
-  onResponderRelease: (event: PressEvent) => void;
-  onResponderTerminate: (event: PressEvent) => void;
-  onResponderTerminationRequest: () => boolean;
-  onStartShouldSetResponder: () => boolean;
-}>;
+};
+
+export type PressableHoverEventProps = {
+  /**
+   * While the user API is onHoverIn the View event is onMouseEnter
+   */
+  onMouseEnter?: (event: MouseEvent) => any;
+
+  /**
+   * While the user API is onHoverOut the View event is onMouseLeave
+   */
+  onMouseLeave?: (event: MouseEvent) => any;
+};
+
+export type PressabilityConfig = Readonly<
+  PressablePressProps &
+    PressableFocusProps &
+    PressableHoverProps & {
```

Comment:
Is this not a flow type that will not get read by typescript? That's where I thought `ReadOnly<>` decorators came from.

Reply:
Is there any chance of this config getting out of sync with RN-Core in a future version? I.E is this something we could just import from RN-Core?

Reply:
TypeScript has a Readonly concept as well. I put it in to PressabilityConfig when I did the port for consistency and am just staying consistent with what I had before here. In terms of getting out of sync, most of the types in here are not exposed and are on the inner Pressability library, not Pressable. As we get rid of the old code the interfaces should shift to be based off of what is exposed on PressableProps as much as possible.